src/longhorn_rpc_client.c

```diff
@@ -45,16 +45,28 @@ void update_timeout_timer(struct lh_client_conn *conn) {
         its.it_interval.tv_sec = 0;
         its.it_interval.tv_nsec = 0;
 
+        // When there are messages on the queue, make timer that expires
+        // request_timeout_period seconds from now.
         if (head_msg != NULL) {
-                its.it_value.tv_sec = head_msg->expiration.tv_sec;
-                its.it_value.tv_nsec = head_msg->expiration.tv_nsec;
+                if (clock_gettime(CLOCK_MONOTONIC, &its.it_value) < 0) {
+                        perror("Fail to get current time");
+                        return;
+                }
+
+                its.it_value.tv_sec += request_timeout_period;
+
                 // Arm
                 if (timerfd_settime(conn->timeout_fd,
                                 TFD_TIMER_ABSTIME,
                                 &its, NULL) < 0) {
                         errorf("BUG: Fail to set new timer\n");
                 }
+        // When there are no more messages on the queue, we disarm
+        // the timer.  This is because we rearm the timer for every
+        // response received; when there are no more messages on the 
+        // queue, there is no response to be received.
```

Comment:
Move this block of comment after `else`, otherwise, it looks like commenting on the previous block.

Reply:
I moved it and indented it.